astropy/modeling/rotations.py

```diff
@@ -443,7 +443,7 @@ class Rotation2D(Model):
 
     _separable = False
 
-    angle = Parameter(default=0.0, getter=_to_orig_unit, setter=_to_radian)
+    angle = Parameter(default=0.0, getter=_to_orig_unit, setter=_to_radian, description="Angle of rotation (if float it should be in deg)")
```

Comment:
```suggestion
    angle = Parameter(default=0.0, getter=_to_orig_unit, setter=_to_radian, description="Angle of rotation (Quantity or value in deg)")
```
I _think_ all `Parameter` accept `Quantity` input (the docs e.g. in `projections` are not entirely clear about this), so I'd suggest to follow this style in general – unless it's considered too lengthy.

Reply:
I like @dhomeier suggestion. If the docs are not clear if a parameters accepts a quantity, then @Akshat1Nar can just create a model and try. If the docs and not clear about it, then we should improve the docs by putting that information in the parameter description!

Reply:
The ones I looked at correctly state `float`; I just was not sure if the docs had not caught up with the wider introduction of quantities. In case of the `Pix2Sky*` etc. models it takes an actual call to `evaluate` to find out that a quantity raises, so it may be safe enough to assume the current docstrings are correct...
Generally it would certainly be preferable if all models could handle quantities of matching type, but that's a project of its own.